src/jdk.httpserver/share/classes/sun/net/httpserver/UnmodifiableHeaders.java

```diff
@@ -32,73 +32,96 @@
+        this.unmodifiableView = Collections.unmodifiableMap(unmodHeaders);
         this.headers = unmodHeaders;
     }
 
+    @Override
     public int size() {return headers.size();}
 
+    @Override
     public boolean isEmpty() {return headers.isEmpty();}
 
+    @Override
     public boolean containsKey(Object key) { return headers.containsKey(key); }
 
+    @Override
     public boolean containsValue(Object value) { return headers.containsValue(value); }
 
+    @Override
     public List<String> get(Object key) { return headers.get(key); }
 
+    @Override
     public String getFirst(String key) { return headers.getFirst(key); }
 
+    @Override
     public List<String> put(String key, List<String> value) {
         throw new UnsupportedOperationException ("unsupported operation");
     }
 
+    @Override
     public void add(String key, String value) {
         throw new UnsupportedOperationException ("unsupported operation");
     }
 
+    @Override
     public void set(String key, String value) {
         throw new UnsupportedOperationException ("unsupported operation");
     }
 
+    @Override
     public List<String> remove(Object key) {
         throw new UnsupportedOperationException ("unsupported operation");
     }
 
+    @Override
     public void putAll(Map<? extends String,? extends List<String>> t)  {
         throw new UnsupportedOperationException ("unsupported operation");
     }
 
+    @Override
     public void clear() {
         throw new UnsupportedOperationException ("unsupported operation");
     }
 
-    public Set<String> keySet() { return map.keySet(); }
+    @Override
+    public Set<String> keySet() { return unmodifiableView.keySet(); }
 
-    public Collection<List<String>> values() { return map.values(); }
+    @Override
+    public Collection<List<String>> values() { return unmodifiableView.values(); }
 
-    /* TODO check that contents of set are not modifable : security */
-
-    public Set<Map.Entry<String, List<String>>> entrySet() { return map.entrySet(); }
+    @Override
+    public Set<Map.Entry<String, List<String>>> entrySet() { return unmodifiableView.entrySet(); }
 
+    @Override
     public List<String> replace(String key, List<String> value) {
         throw new UnsupportedOperationException("unsupported operation");
     }
 
+    @Override
     public boolean replace(String key, List<String> oldValue, List<String> newValue) {
         throw new UnsupportedOperationException ("unsupported operation");
     }
 
+    @Override
     public void replaceAll(BiFunction<? super String, ? super List<String>, ? extends List<String>> function) {
         throw new UnsupportedOperationException ("unsupported operation");
     }
 
+    @Override
     public boolean equals(Object o) {return headers.equals(o);}
 
+    @Override
     public int hashCode() {return headers.hashCode();}
+
+    @Override
+    public String toString() {
+        return this.headers.toString();
```

Comment:
```suggestion
        return headers.toString();
```
to keep with the style of other methods in this class